control_plane/src/storage.rs

```diff
@@ -235,16 +234,27 @@ impl PageServerNode {
             args.extend(["-c", config_override]);
         }
 
-        let mut cmd = Command::new(self.env.pageserver_bin()?);
-        let mut filled_cmd = fill_rust_env_vars(cmd.args(&args).arg("--daemonize"));
-        filled_cmd = fill_aws_secrets_vars(filled_cmd);
-
-        if !filled_cmd.status()?.success() {
-            bail!(
+        let background_command = subprocess::Exec::cmd(self.env.pageserver_bin()?)
+            .stdout(subprocess::NullFile)
+            .stderr(subprocess::NullFile)
+            .args(&args)
+            // do not wait for the subprocess to exit when it gets dropped: we write its pid into a file
+            // to retrieve and manage on another CLI restart
+            .detached();
+        let filled_cmd = fill_aws_secrets_vars!(fill_rust_env_vars!(background_command));
+
+        // by default, `subprocess::Popen` waits for process to complete on drop
+        let mut pageserver_subprocess = filled_cmd.popen().with_context(|| {
+            format!(
                 "Pageserver failed to start. See console output and '{}' for details.",
                 datadir.join("pageserver.log").display()
-            );
-        }
+            )
+        })?;
+        let pageserver_pid = pageserver_subprocess
+            .pid()
+            .context("Pageserver subprocess got no pid")?;
+        std::fs::write(&pid_file_path, pageserver_pid.to_string())
+            .context("Failed to write pageserver pid file")?;
 
```

Comment:
race condition: if you launch two instances at the same time, one can overwrite the pid file of the other, and you end up with two pageservers running in the same data directory.

Reply:
This one is easy to fix, using the `OpenOptions` and require to check for the new file creation there.